Documentation/Basic-Usage.md

```diff
@@ -1,44 +1,58 @@
-Once you have the `AnyObject`, you can call the global `decode` function to get
-back the decoded model.
+Argo uses Swift's type system along with concepts from functional programming to
+let you smoothly transform JSON data into Swift model objects or structs. Argo's
+approach does this with a minimum of syntax, while at the same time improving
+type safety and data integrity compared to other approaches.
+
+You may need to learn a few things in order to learn Argo effectively, but once
+you do so, you'll have a powerful new tool to hang on your belt!
+
+### Decoding basics
+
+Argo's whole purpose is to let you easily pick apart structured data (normally
+in the form of a dictionary created from JSON data) and create Swift objects or
+structs based on the decoded content. Typically, you'll want to do this with
+JSON data received from a server or elsewhere. The first thing you need to do is
+convert the JSON data from `NSData` to an `AnyObject` using `Foundation`'s
+`NSJSONSerialization` API.  Once you have the `AnyObject`, you can call Argo's
+global `decode` function to get back the decoded model.
 
 ```swift
 let json: AnyObject? = try? NSJSONSerialization.JSONObjectWithData(responseData, options: [])
 
 if let j: AnyObject = json {
-  let user: User? = decode(j) // ignore error info or
-  let decodedUser: Decoded<User> = decode(j) // preserve error info
+  let user: User? = decode(j)                  // ignore failure info or
+  let decodedUser: Decoded<User> = decode(j)   // preserve failure info
 }
 ```
 
-Argo 1.0 introduces a new type: `Decoded<T>`. This is the type returned from
-the `decode` function that you implement as part of the `Decodable` protocol.
-This new type allows you to preserve information about why a decoding failed.
-You can choose to either ignore the `Decoded` type and just get back the
-optional value or keep the `Decoded` type and use it to debug decoding errors.
-When you decode an `AnyObject` into a model using the global `decode` function,
-you can specify whether you want an `Optional` model or a `Decoded` model by
-specifying the return type as seen in the code block above.
+As you see in this example, Argo introduces a new type: `Decoded<T>`.  This new
+type contains either a successfully decoded object or a failure state that
+preserves information about why a decoding failed. You can choose to either
+ignore the `Decoded` type and just get back the optional value or keep the
+`Decoded` type and use it to debug or report decoding failures.  When you decode
+an `AnyObject` into a model using the global `decode` function, you can specify
+whether you want an `Optional` model or a `Decoded` model by specifying the
+return type as seen in the code block above.
+
+### Implementing `Decodable`
 
-Next, you need to make sure that models that you wish to decode from JSON
-conform to the `Decodable` protocol:
+In order for this to work with your own model classes or structs, you need to make
```

Comment:
I think we should remove `classes or structs` here and go back to `models`. Or possibly say `model types`. Argo works with `enum`s too, I don't want us to give the impression that it doesn't.